07-routing-gossip.md

```diff
@@ -559,13 +559,34 @@ are unique 8-byte values, no more than 14 bytes can be duplicated
 across the stream: as each duplicate takes at least 2 bits, no valid
 contents could decompress to more then 3669960 bytes.
 
+Query messages can be extended with optional fields that can help reduce the number of messages needed to synchronize routing tables by enabling:
+
+- timestamp-based filtering of `channel_update` messages: don't ask for `channel_update` messages that
+ are older than the ones you already have.
+- checksum-based filtering of `channel_update` messages: don't ask for `channel_update` messages that carry the same information as the ones you already have.
+
 ### The `query_short_channel_ids`/`reply_short_channel_ids_end` Messages
 
 1. type: 261 (`query_short_channel_ids`) (`gossip_queries`)
 2. data:
     * [`32`:`chain_hash`]
     * [`2`:`len`]
     * [`len`:`encoded_short_ids`]
+    * [`query_short_channel_ids_tlv`]
+      * type: 1 (`query_flags`)
+      * data:
+        * [`1`:`encoding_type`]
+        * [`len-1`:`encoded_query_flags`]
```

Comment:
`encoded_query_flags` is a binary blob of size `len - 1`. If `encoding_type` is zero then yes its size should be `num_sids`, but if its one then you need to decompress it first, then the length of the decompressed buffer should be `num_sids`. I'll rephrase and make this more explicit

Reply:
this length might also be inaccurate if we are encoding them as varints rather than bytes, but tbh i don't have a better suggestion on how to describe the expected length :P